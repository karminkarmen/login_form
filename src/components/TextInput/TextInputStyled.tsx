import styled from 'styled-components';

type Props = {
    label: string, 
    type: string
};

const InputStyled = styled.input.attrs((props: Props) => ({
    type: props.type,
    name: props.label,
    id: props.label
}))<Props>`
    font-size: ${props => props.theme.fontSizes.medium};
    border-radius: ${props => props.theme.radius.medium};
    border: 2px solid ${props => props.theme.colors.main};
    padding: ${props => props.theme.paddings.medium};
    background: transparent;
    color: ${props => props.theme.colors.secondaryLightest};
    margin-bottom: 1rem;

    ::placeholder {
        color: ${props => props.theme.colors.background};
    }

    &:focus {
        outline: none;
        background: ${props => props.theme.colors.secondaryLighten};
        color: ${props => props.theme.colors.main};
    }
`;

export default InputStyled;