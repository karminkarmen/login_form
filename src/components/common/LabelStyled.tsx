import styled from 'styled-components';

type Props = {
    target: string;
    isUppercased: boolean;
}

const LabelStyled = styled.label.attrs((props: Props) => ({
    htmlFor: props.target
}))<Props>`
    margin: ${props => props.theme.margins.normal};
    color: ${props => props.theme.colors.main};
    text-transform: ${props => props.isUppercased ? 'uppercase' : 'none'};
    font-weight: bold;
    letter-spacing: ${props => props.theme.letterSpacing.normal};
    font-size: ${props => props.theme.fontSizes.normal};
    font-family: ${props => props.theme.fontFamilies.main}, sans-serif;
`;

export default LabelStyled;
