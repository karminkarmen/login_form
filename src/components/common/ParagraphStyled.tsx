import styled from 'styled-components';

const ParagraphStyled = styled.p`
    margin: 0 0 ${props => props.theme.margins.normal} 0;
    max-width: ${props => props.theme.width.innerElementWidth};
    color: ${props => props.theme.colors.main};
    font-size: ${props => props.theme.fontSizes.normal};
    font-family: ${props => props.theme.fontFamilies.lato};
`;

export default ParagraphStyled;