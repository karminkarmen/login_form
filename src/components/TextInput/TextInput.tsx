import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import TextInputStyled from './TextInputStyled';
import LabelStyled from '../common/LabelStyled';
import ParagraphStyled from '../common/ParagraphStyled';

type Props = {
    handler: (e: React.ChangeEvent<HTMLInputElement>) => void;
    label: string;
    type: string;
    validationInfo: string;
}

const TextInput: React.FC<Props> = ({ handler, label, type, validationInfo }) => {
    return (
        <ThemeProvider theme={theme}>
            <LabelStyled 
                target={label}
                isUppercased={true}
            >
                {label}
            </LabelStyled>
            <TextInputStyled 
                onChange={handler} 
                type={type} 
                label={label}
            />
            {validationInfo.length > 0 && <ParagraphStyled>{validationInfo}</ParagraphStyled>} 
        </ThemeProvider>
    );
};

export default TextInput;