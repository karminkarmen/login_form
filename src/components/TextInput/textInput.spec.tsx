import React from 'react';
import { shallow } from 'enzyme';

import TextInput from './TextInput';
import LabelStyled from '../common/LabelStyled';
import TextInputStyled from './TextInputStyled';
import ParagraphStyled from '../common/ParagraphStyled';

describe('<CheckboxInput />', () => {

    const props: React.ComponentProps<typeof TextInput> = {
        handler: (e: React.ChangeEvent<HTMLInputElement>) => {},
        label: 'login',
        type: 'password',
        validationInfo: 'Validation info',
    };

    it('matches snapshot', () => {
        const formattingButtons = shallow(<TextInput {...props} />);
        expect(formattingButtons.debug()).toMatchSnapshot();
    });

    it('should render styled label', () => {
        const wrapper = shallow(<TextInput {...props} />);

        expect(wrapper.find(LabelStyled)).toHaveLength(1);
    });

    it('should render styled label connected with the input', () => {
        const wrapper = shallow(<TextInput {...props} />);

        const label = wrapper.find(LabelStyled);
        expect(label.find({ target: props.label })).toHaveLength(1);
    });

    it('should render styled label which contains provided text', () => {
        const wrapper = shallow(<TextInput {...props} />);

        const label = wrapper.find(LabelStyled);
        expect(label.text()).toEqual(props.label);
    });

    it('should render styled input', () => {
        const wrapper = shallow(<TextInput {...props} />);

        expect(wrapper.find(TextInputStyled)).toHaveLength(1);
    });

    it('should render styled paragraph', () => {
        const wrapper = shallow(<TextInput {...props} />);

        expect(wrapper.find(ParagraphStyled)).toHaveLength(1);
    });

    it('should render styled paragraph with error info if provided', () => {
        const wrapper = shallow(<TextInput {...props} />);

        const label = wrapper.find(ParagraphStyled);
        expect(label.text()).toEqual(props.validationInfo);
    });

    it('should not render styled paragraph with error info if it is not provided', () => {
        props.validationInfo = '';
        const wrapper = shallow(<TextInput {...props} />);

        expect(wrapper.find(ParagraphStyled)).toHaveLength(0);
    });
});
