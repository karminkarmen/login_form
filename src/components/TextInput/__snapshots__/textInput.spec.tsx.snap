// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxInput /> matches snapshot 1`] = `
"<ThemeProvider theme={{...}}>
  <styled.label target=\\"login\\" isUppercased={true}>
    login
  </styled.label>
  <styled.input onChange={[Function: handler]} type=\\"password\\" label=\\"login\\" />
  <styled.p>
    Validation info
  </styled.p>
</ThemeProvider>"
`;
